cookbook/src/components/forms/new-cookbook-form/newCookbookForm.js

```diff
@@ -0,0 +1,124 @@
+import {closeModalAction} from "../../../redux/actions/modal-window/modalWindowActions";
+import {
+    imageUploadFail,
+    resetImageData,
+    startImageUpload
+} from "../../../redux/actions/image-storage/imageStorageActions";
+import {useDispatch, useSelector} from "react-redux";
+import {
+    addCookbookDescription, addCookbookImage,
+    addCookbookTitle,
+    addCookbookUsername, addRecipe, resetCookbookData
+} from "../../../redux/actions/new-cookbook/newCookbookActions";
+import RecipeTabsList from "../../page-parts/recipe-tabs-list/RecipeTabsList";
+import {startCookbookUpload} from "../../../redux/actions/data-upload/dataUploadActions";
+import {fetchUserCookbooksStart} from "../../../redux/actions/data-fetch/dataFetchActions";
+
+const NewCookbookForm = () => {
+    const dispatch = useDispatch();
+
+    const recipes = useSelector(state => state.dataFetchReducer.recipes.userRecipes);
+    const userName = useSelector(state => state.userDataReducer.user.name && state.userDataReducer.user.name);
+    const currentUserId = useSelector(state => state.auth.currentUser.uid && state.auth.currentUser.uid);
+    const imageUploadError = useSelector(state => state.imageStorageReducer.error && state.imageStorageReducer.error);
+    const message = useSelector(state => state.imageStorageReducer.message && state.imageStorageReducer.message);
+    const imageURL = useSelector(state => state.imageStorageReducer.imageURL && state.imageStorageReducer.imageURL);
+    const selectedRecipesIDs = useSelector(state => state.newCookbookReducer.recipes);
+    const selectedRecipes = recipes.filter(recipe => selectedRecipesIDs.indexOf(recipe.recipeId) !== -1);
```

Comment:
You can manage selected recipes in component state
Why do you use redux here?

Reply:
Selected recipes are used not only in the component but later are sent to the firebase